src/backend/aspen/cli/sync_auth0.py

```diff
@@ -0,0 +1,420 @@
+        token = auth_req.client_credentials(
+            client_id, client_secret, f"https://{domain}/api/v2/"
+        )
+
+        self.client = Auth0(domain=domain, token=token["access_token"])
+
+    def get_all_results(self, endpoint, key):
+        # Auth0 paginates results. We don't have a crazy amount of data in auth0 so we can
+        # afford to just paginate through all the results and hold everything in memory.
+        results = []
+        page = 0
+        per_page = 25
+        while True:
+            resp = endpoint(page=page, per_page=per_page)
+            last_result = resp["start"] + resp["limit"]
+            results.extend(resp[key])
+            if last_result >= resp["total"]:
+                return results
+            page += 1
+
+    def get_users(self):
+        return self.get_all_results(self.client.users.list, "users")
+
+    def get_orgs(self):
+        return self.get_all_results(
+            self.client.organizations.all_organizations, "organizations"
+        )
+
+    @cache
+    def get_auth0_role(self, role_name):
+        all_roles = self.get_roles()
+        for role in all_roles:
+            if role["name"] == role_name:
+                return role
+
+    @cache
+    def get_roles(self):
+        return self.get_all_results(self.client.roles.list, "roles")
+
+    def get_org_members(self, org):
+        return self.get_all_results(
+            partial(self.client.organizations.all_organization_members, org["id"]),
+            "members",
+        )
+
+    def add_org_member(self, org, user_id):
+        self.client.organizations.create_organization_members(
+            org["id"], {"members": [user_id]}
+        )
+        member_role = self.get_auth0_role("member")
+        self.client.organizations.create_organization_member_roles(
+            org["id"], user_id, {"roles": [member_role["id"]]}
+        )
+
+    def remove_org_member(self, org: Auth0Org, user_id: int):
+        self.client.organizations.delete_organization_members(
+            org["id"], {"members": [user_id]}
+        )
+
+    def add_org(self, group_id: int, org_name: str):
+        body = {
+            "name": f"group-{group_id}",
+            "display_name": org_name,
+        }
+        self.client.organizations.create_organization(body)
+
+    def delete_org(self, org_id: int):
+        self.client.organizations.delete_organization(org_id)
+
+    def create_user(self, email, name):
+        body = {
+            "email": email,
+            "user_metadata": {},
+            "email_verified": True,
+            "name": name,
+            "verify_email": False,
+            "password": generate_password(),
+            "connection": "Username-Password-Authentication",
+        }
+        self.client.users.create(body)
+
+    def delete_user(self, auth0_user_id):
+        self.client.users.delete(auth0_user_id)
+
+
+def compare_stuff(auth0_objects, db_objects, auth0_match_field, db_match_field):
```

Comment:
very technical name 😆

Reply:
yeah it's cute, but if it's comparing user data I'd like the name be more reflective of it

Reply:
Yep, updated!